src/components/Card.js

```diff
@@ -27,17 +31,51 @@ export default class Card {
     this._cardImageElement.alt = this.name;
     this._likeButton = this._element.querySelector(".card__like-button");
     this._deleteButton = this._element.querySelector(".card__delete-button");
-
+    this._checkLikeStatus();
     this._setEventListeners();
     return this._element;
   }
 
   _handleLikeButton() {
     this._likeButton.classList.toggle("card__like-button_active");
+    if (this.isLiked) {
+      api.removeLike(this._id).catch((err) => console.log(err));
+      this.isLiked = false;
+    } else {
+      api.addLike(this._id).catch((err) => console.log(err));
+      this.isLiked = true;
+    }
   }
 
   _handleDeleteButton() {
-    this._element.remove();
+    this._openDeteleConfirmPopup();
+  }
+
+  _checkLikeStatus() {
+    if (this.isLiked) {
+      this._likeButton.classList.add("card__like-button_active");
+    } else {
+      this._likeButton.classList.remove("card__like-button_active");
+    }
+  }
+
+  _openDeteleConfirmPopup() {
+    const deletePopup = new PopupDeleteConfirm("#delete-card-modal", () => {
+      this._handleConfirmDelete();
+    });
```

Comment:
Every popup should be created only one time in `index.js` otherwise there is a waste of resources on searching elements again and again inside the class